snuba_sdk/dsl/dsl.py

```diff
@@ -0,0 +1,296 @@
+"""
+Contains the definition of MQL, the Metrics Query Language.
+Use `parse_mql()` to parse an MQL string into a MetricsQuery.
+"""
+
+from typing import Any, Mapping, Sequence
+
+from parsimonious.exceptions import ParseError
+from parsimonious.grammar import Grammar
+from parsimonious.nodes import Node, NodeVisitor
+
+from snuba_sdk.arithmetic import ArithmeticFunction
+from snuba_sdk.column import Column, Variable
+from snuba_sdk.conditions import Condition, Op
+from snuba_sdk.function import Function
+from snuba_sdk.metrics_query import MetricsQuery
+from snuba_sdk.query_visitors import InvalidQueryError
+from snuba_sdk.timeseries import Metric, Timeseries
+
+
+AGGREGATE_PLACEHOLDER_NAME = "AGGREGATE_PLACEHOLDER"
+ENTITY_TYPE_REGEX = r"(c|s|d|g|e)"
```

Comment:
I know that right now these values line up with Entities, but in the future they might not. Really these are metric types, not entity types.

Reply:
Renamed to `METRIC_TYPE_REGEX`